### ISSUE TITLE
Improve get_vector_double

### ISSUE THREAD

**Issue by bangerth** [geodynamics/aspect#528]
`multicomponent.cc` has a curious function `get_vector_double()` which has now also been copied to `diffusion_dislocation.cc` in #502. We should come up with a better name for the function and maybe move it to `utilities.h` or similar.

**Comment by bobmyhill** [comment]
Yes, I added this function to shorten the very long parameter parsing section of multicomponent.cc. It's the vector (or rather, comma delimited list) equivalent of get_double. get_double_list or get_list_of_doubles, perhaps?

utilities.h would be a good place for the function, I think. Should the function itself be modified?

**Comment by bangerth** [comment]
Let's not worry about this for #502 itself, but deal with it as a follow-up.

My preference would be to have a function of the kind
```
  /**
   * Given an array @p values, consider three cases:
   * - If it has size @p N, return the original array.
   * - If it has size one, return an array of size @p N where all
   *   elements are equal to the one element of @p value.
   * - If it has any other size, throw an exception that uses
   *   @p id_text as an identifying string.
   *
   * This function is typically used for parameter lists that can either
   * contain different values for each of a set of objects (e.g., for
   * each compositional field), or contain a single value that is then
   * used for each object.
   */
  template <typename T>
  std::vector<T>
  possibly_extend_from_1_to_N (const std::vector<T> &values,
                                                     const unsigned int N,
                                                     const std::string &id_text); 
```
The function would then be called as in
```
  densities
   =possibly_extend_from_1_to_N(Utilities::string_to_double(Utilities::split_string(prm.get("Densities"))),
                                                      n_fields,
                                                      "Material model/Diffusion dislocation/Densities");
```

**Comment by class4kayaker** [comment]
Implementation in progress

**Comment by gassmoeller** [comment]
Fixed by #1030
